### PR TITLE
[TIMOB-25600] Android : SDK 7.0.0.GA not applying custom theme attributes properly on TableViewRow, but SDK 5.5.1 or below works properly

### PR DESCRIPTION
**Pull request by ypbnv**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25600

**Description:**
Rework message handling for TableView to not prematurely create the native view and thus not taking theme properties into account.
Change context and property used for applyTouchFeedback method. It now uses the view's context and allows colors directly defined in theme's properties.

**Test case:**
**app.js**
```
var win = Ti.UI.createWindow({
	layout: 'vertical',
	backgroundColor : 'black',
	theme : 'GalleryTheme'
});

var table = Ti.UI.createTableView({
	width : Ti.UI.FILL,
	height : '90%',
	data: [createRow('Row from creation dictionary')]
});

for (var i=0; i < 20; i++) {
	table.appendRow(createRow('Append no:' + i));
}

table.insertRowAfter(4, createRow('Insert row after'));

table.deleteRow(4);

win.add(table);
win.open();

var buttonContainer = Ti.UI.createView({layout: 'horizontal', width: Ti.UI.SIZE});

var button1 = Ti.UI.createButton({title:'Scroll to 4'});
var button2 = Ti.UI.createButton({title:'Select row 2'});
var button3 = Ti.UI.createButton({title:'Scroll to top'});

buttonContainer.add(button1);
buttonContainer.add(button2);
buttonContainer.add(button3);

button1.addEventListener('click', function() {
	table.scrollToIndex(4);
});

button2.addEventListener('click', function() {
	table.selectRow(2);
});

button3.addEventListener('click', function() {
	table.scrollToTop(0);
});

win.add(buttonContainer);

function createRow(_text) {
	var label = Ti.UI.createLabel({
		color: 'white',
		text: _text,
		left : 20,
		width: Ti.UI.SIZE, height: Ti.UI.SIZE
	});

	var row = Ti.UI.createTableViewRow({
		height: 50,
		width: Ti.UI.FILL
	});

	row.add(label);
	return row;
}
```

**customtheme.xml**
```
<?xml version="1.0" encoding="utf-8"?>
<resources>
	<style name="GalleryTheme" parent="Theme.AppCompat.Light">
		<item name="android:textColorPrimary">#980054</item>
		<item name="colorControlHighlight">#2eba4d</item>
		<item name="colorPrimary">#ffbc77</item>
		<item name="colorPrimaryDark">#cf9458</item>
		<item name="colorAccent">#4064ff</item>
	</style>
</resources>
```

### CODE REVIEW

**Comment by lokeshchdhry** [comment]
FR Passed.

The custom theme is getting applied properly to the `tableViewRow`.

Studio Ver:     5.0.0.201712081732
SDK Ver:        7.2.0 local build
OS Ver:         10.13.2
Xcode Ver:      Xcode 9.2
Appc NPM:       4.2.12
Appc CLI:       7.0.2
Daemon Ver:     1.0.1
Ti CLI Ver:     5.0.14
Alloy Ver:      1.11.0
Node Ver:       8.9.1
NPM Ver:        5.5.1
Java Ver:       1.8.0_101
Devices:        ⇨ google Nexus 5 --- Android 6.0.1
                ⇨ google Nexus 6P --- Android 8.0.0

**Comment by lokeshchdhry** [comment]
Waiting for CR to merge.

File: android/modules/ui/src/java/ti/modules/titanium/ui/TableViewProxy.java [hunk 1]
```diff
@@ -335,12 +339,15 @@ public void deleteRow(Object row, @Kroll.argument(optional = true) KrollDict opt
...
 	private void handleDeleteRow(Object row) throws IllegalStateException
 	{
+		TiUITableView nativeTableViewReference = ((TiUITableView) peekView());
 		if (row instanceof Integer) {
 			int index = (Integer) row;
 			RowResult rr = new RowResult();
 			if (locateIndex(index, rr)) {
 				rr.section.removeRowAt(rr.rowIndexInSection);
-				getTableView().setModelDirty();
+				if (nativeTableViewReference != null) {
```
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/TableViewProxy.java]: Maybe we could put this into a method to prevent duplicate code?

**Review by garymathews — APPROVED** [review]
CR: **PASS**

Although, could make minor improvement.

**Comment by ypbnv** [comment]
@garymathews Updated.

**Pull request merged**